src/rm_topk.c

```diff
@@ -77,28 +77,37 @@ int TopK_Create_Cmd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 }
 
 int TopK_Add_Cmd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
-//    RedisModule_AutoMemory(ctx);
+
     if (argc < 3)
         return RedisModule_WrongArity(ctx);
-
+        
     TopK *topk;
     if(GetTopKKey(ctx, argv[1], &topk,  REDISMODULE_READ | REDISMODULE_WRITE) != 
                             REDISMODULE_OK) {
         return REDISMODULE_OK;
     }
 
-    size_t itemlen;
-    for(int i = 2; i < argc; ++i) {
-        const char *item = RedisModule_StringPtrLen(argv[i], &itemlen);
-        TopK_Add(topk, item, itemlen);
+    int itemCount = argc - 2;
+    RedisModule_ReplyWithArray(ctx, itemCount);
+
+    for(int i = 0; i < itemCount; ++i) {
+        size_t itemlen;
+        const char *item = RedisModule_StringPtrLen(argv[i + 2], &itemlen);
+        char *expelledItem = TopK_Add(topk, item, itemlen);
+  
+        if(expelledItem == NULL) {
+            RedisModule_ReplyWithNull(ctx);
+        } else {
+            RedisModule_ReplyWithSimpleString(ctx, expelledItem);
+            free(expelledItem);
```

Comment:
This was formerly a call to `TOPK_FREE` (`topk.c:148`).

Reply:
@jeffreylovitz , great catch. Thank you.